src/business_logic/state/cached_state.rs

```diff
@@ -80,10 +82,14 @@ impl<T: StateReader + Clone> StateReader for CachedState<T> {
 
     fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<&[u8; 32], StateError> {
         if self.cache.get_class_hash(contract_address).is_none() {
-            let class_hash = self.state_reader.get_class_hash_at(contract_address)?;
+            let class_hash = match self.state_reader.get_class_hash_at(contract_address) {
+                Ok(x) => Cow::Borrowed(x),
```

Comment:
One last thing, I think it is unnecessary to add this Cow, it adds doesn't make a significant improvement in the performance, since we end up cloning the value.
From my point of view, I think it will be better to delete this Cow.